samples/snippets/src/main/java/com/example/spanner/AlterTableWithForeignKeyDeleteCascadeSample.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_alter_table_with_foreign_key_delete_cascade]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Instance;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.InstanceConfigId;
+import com.google.cloud.spanner.InstanceId;
+import com.google.cloud.spanner.InstanceInfo;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.common.collect.ImmutableList;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import com.google.spanner.admin.instance.v1.CreateInstanceMetadata;
+import java.util.concurrent.ExecutionException;
+
+class AlterTableWithForeignKeyDeleteCascadeSample {
+
+  static void alterForeignKeyDeleteCascadeConstraint() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseAdminClient adminClient = spanner.getDatabaseAdminClient();
+      alterForeignKeyDeleteCascadeConstraint(adminClient, instanceId, databaseId);
+    }
+  }
+
+  static void alterForeignKeyDeleteCascadeConstraint(
+      DatabaseAdminClient adminClient, String instanceId, String databaseId) {
+    adminClient.updateDatabaseDdl(
+            instanceId,
+            databaseId,
+            ImmutableList.of("ALTER TABLE ShoppingCarts\n"
+                + "              ADD CONSTRAINT FKShoppingCartsCustomerName\n"
+                + "              FOREIGN KEY (CustomerName)\n"
+                + "              REFERENCES Customers(CustomerName)\n"
```

Comment:
Is this something that has been standardized across multiple client libraries? I am a little surprised that our sample uses `CustomerName` as the referenced column, as it requires the column to be unique. `CustomerName` does not sound like a column that would normally be unique.

Reply:
Yes I agree on this. And unfortunately this has been standardised across all client libraries.

Reply:
Ack, thanks for checking.